files/image_config/config-chassis/config-chassis.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# 
+###########################################################################
+# config-chassis.py
+# 
+# Chassis preconfiguration service.
+# This module provides preconfiguration service for chassis based system
+# based on data provided by
+# /use/share/sonic/devices/<platform>/chassis.yaml file
+#
+###########################################################################
+# Copyright (c) 2021 Cisco Systems, Inc. and its affiliates.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+###########################################################################
+
+import argparse
+import logging
+import os.path
+import subprocess
+import yaml
+
+ETC_HOSTS_FILE = '/etc/hosts'
+HTTP_CONFIG_FILE = '/etc/sonic/http_config.yaml'
+
+class ChassisConfig():
+    def __init__(self, args, **kw):
+        self.chassis_conf = None
+        self.http_config = None
+        if args.config_file:
+            chassis_yaml = args.config_file
+        else:
+            # get chassis.yaml path based on platform type
+            self.get_platform()
+            chassis_yaml=os.path.join('/usr/share/sonic/device', self.platform, 'chassis.yaml')
+
+        if os.path.isfile(chassis_yaml):
+            with open(chassis_yaml) as f:
+                self.chassis_conf = yaml.safe_load(f)
+
+            # Extract http config
+            self.http_config = self.chassis_conf.get('http_config')
+        else:
+            logger.info('chassis config file {} not present - skip config'.format(chassis_yaml))
+
+    def get_platform(self):
+        ''' Get the platform type from DEVICE_METADATA '''
+        SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+        PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
```

Comment:
On a system before the config\minigraph is loaded, `'DEVICE_METADATA.localhost.platform'` may not be available

Reply:
sonic-cfggen call get_platform which fetch the platform from machine.conf. -H flag helps to construct DEVICE_METADATA info. So we should be good. database.sh use same command to get platform info which is before config is loaded.